connectivity/check/action.go

```diff
@@ -163,23 +167,24 @@ func (a *Action) ExecInPod(ctx context.Context, cmd []string) {
 	cmdName := cmd[0]
 	cmdStr := strings.Join(cmd, " ")
 
-	if stderr.Len() > 0 {
-		a.test.Debugf("%s stderr: %s", cmdName, stderr.String())
-	} else if stdout.Len() > 0 {
-		a.test.Debugf("%s stdout: %s", cmdName, stdout.String())
-	}
-
 	if err != nil || stderr.Len() > 0 {
 		if a.shouldSucceed() {
 			a.Failf("command %q failed: %s", cmdStr, err)
 		} else {
-			a.test.Debugf("command %q failed as expected: %s", cmdStr, err)
+			a.test.Infof("command %q failed as expected: %s", cmdStr, err)
 		}
 	} else {
 		if !a.shouldSucceed() {
 			a.Failf("command %q succeeded while it should have failed: %s", cmdStr, stdout.String())
 		}
 	}
+	if a.failed {
+		if stderr.Len() > 0 {
+			a.Warnf("%s stderr: %s", cmdName, stderr.String())
```

Comment:
`Failf()` sounds good to me ✅

Reply:
I was thinking the exit code of the execution would be sufficient to indicate the success/fail status of the command. No trouble to make this a `Failf()` though, will add a comment that we consider any `stderr` output to indicate a failing test as well.

Reply:
Was considering if we'd still need to log these at debug level if the action did not fail, though. Thoughts?

Reply:
Changed this to log stderr and stdout at fail level if the action failed (i.e., command succeeded when it should not have, or failed when it should have succeeded), and at debug level otherwise (to always see the output, if any, with `--debug`).